src/app/portlets/dot-edit-page/main/dot-edit-page-main/dot-edit-page-main.component.ts

```diff
@@ -1,19 +1,56 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
-import { Observable } from 'rxjs/Observable';
 import { DotRenderedPageState } from '../../shared/models/dot-rendered-page-state.model';
+import { DotContentletEditorService } from '../../../../view/components/dot-contentlet-editor/services/dot-contentlet-editor.service';
+import { Observable } from 'rxjs/Observable';
+import { Subject } from 'rxjs/Subject';
+import { takeUntil } from 'rxjs/operators';
+import { DotPageStateService } from '../../content/services/dot-page-state/dot-page-state.service';
 
 @Component({
     selector: 'dot-edit-page-main',
     templateUrl: './dot-edit-page-main.component.html',
     styleUrls: ['./dot-edit-page-main.component.scss']
 })
-export class DotEditPageMainComponent implements OnInit {
+export class DotEditPageMainComponent implements OnInit, OnDestroy {
     pageState: Observable<DotRenderedPageState>;
+    private destroy$: Subject<boolean> = new Subject<boolean>();
 
-    constructor(private route: ActivatedRoute) {}
+    constructor(
+        private route: ActivatedRoute,
+        private dotContentletEditorService: DotContentletEditorService,
+        private dotPageStateService: DotPageStateService
+    ) {}
 
     ngOnInit() {
         this.pageState = this.route.data.pluck('content');
+        this.subscribeIframeCloseAction();
+    }
+
+    ngOnDestroy(): void {
+        this.destroy$.next(true);
+        this.destroy$.complete();
+    }
+
+    /**
+     * Handle call of contentlet dialog
+     *
+     * @param {inode: string; label: string} buttonClicked
+     * @memberof DotEditPageMainComponent
+     */
+    openPageProperties(buttonClicked: { inode: string; label: string }) {
```

Comment:
This is too tight up, if we add another button to the nav how do we handle it? We need to listen for the action from the nav and base on that do something, here we're assuming that always we're opening page properties.